### DISCUSSION TITLE
How I can use  formality?

### DISCUSSION THREAD

**Discussion by AlexandraKartashovaFusion** [Fekide/strapi-plugin-translate#107] [Q&A]
Hi Guys.
Could you tell me how to connect the packages  strapi-plugin-deepl  & strapi-provider-translate-deepl for use formality, please.
Is it possible to make a demo project?

**Comment by layaxx** [comment]
Hi, 
we do have a demo project set up [in the playground folder.](https://github.com/Fekide/strapi-plugin-translate/tree/main/playground)
If you want to override the default formality, you would need to update the plugin config (`playground/config/plugins.js`) to something like this:
```js
module.exports = ({ env }) => ({
  // ...
  translate: {
    enabled: true,
    config: {
      provider: "deepl",
      providerOptions: {
        apiOptions: {
          formality: "prefer_more",
        },
      },
    },
  },
  // ...
})
```

Also make sure to set your Deepl API Key, either via this config file or via environment variables, see `playground/.env.example` for reference.

Hope this helps :)

**Comment by AlexandraKartashovaFusion** [comment]
Hi,
Where in Network I can see that the request was sent with "formality"?

**Reply by sargreal** [reply]
This is currently only a server side thing, so it will not show up in the network tab. The formality should however be applied at each request that is sent to the DeepL-API. If that is not the case it might be a bug.
In a upcoming release (probably 1.2) it will also be possible to set these options in the modal before translation.

**Reply by sargreal** [reply]
All possible values for apiOptions are listed here and should be respected. formality also seems correct:
https://github.com/DeepLcom/deepl-node#text-translation-options

**Reply by layaxx** [reply]
Pretty much correct, `tagHandling` is the only option that cannot be overriden this way as it would break markdown translation. 
But I just checked and formality works as expected, at least when translating english -> german

**Comment by AlexandraKartashovaFusion** [comment]
Hi, guys
I have this error for the most part of my pages
{
    "status": 500,
    "name": "InternalServerError",
    "message": "CMEditViewTranslateLocale.translate-failure",
    "details": {
      "message": "Undefined binding(s) detected when compiling WHERE. Undefined column(s): [t0.id] query: where \"t0\".\"id\" = ?"
    }
}

package.json
{
    "@strapi/helper-plugin": "^4.6.1",
    "@strapi/plugin-i18n": "4.6.1",
    "@strapi/plugin-users-permissions": "4.6.1",
    "@strapi/strapi": "4.6.1",
    "strapi-plugin-populate-deep": "^0.1.2",
    "strapi-plugin-translate": "1.0.0-next.2",
    "strapi-provider-translate-deepl": "1.0.0-next.2"
}

**Reply by layaxx** [reply]
Hi, 
I dont have time to investigate this atm but you could try if it works with the latest versions, i.e. [strapi-provider-translate-deepl@1.1.0](https://github.com/Fekide/strapi-plugin-translate/releases/tag/strapi-provider-translate-deepl%401.1.0) and [strapi-plugin-translate@1.1.0](https://github.com/Fekide/strapi-plugin-translate/releases/tag/strapi-plugin-translate%401.1.0)

**Reply by sargreal** [reply]
So there are definitely some issues here:

1. Did you setup the playground exactly as it is there also with the cloned repository? There are some issues right now with the dependencies there which might cause your problems, but also the plugin and provider are out of date, see next point
2. As @layaxx said, your versions of `strapi-plugin-translate` and `strapi-provider-translate-deepl` are out of date, please use for both the version `1.1.0` so we can make sure that your issue is not because of a wrong version. That version problem might also be the cause of why your setting for `formality` does not work
3. If the previous does not already fix your issues, please give us some additional information:
   1. How did you translate? Most likely direct on a single entry right?
   2. Can you show us the content type schema that you used that produced this error? It might be connected with relation translation.
   3. Since only the schema might not be enough to , could you give us an example of how your entity is filled? Are the relations empty or are the filled with one or multiple entries?

**Reply by AlexandraKartashovaFusion** [reply]
Update to the version 1.1.0 helped me
